Runner/CustomTemplateFactory.py

```diff
@@ -0,0 +1,170 @@
+import json
+"""
+This module responsiblilty is for reading and setting elements of the json templates
+"""
+
+
+def set_template_name(template, pool_id):
+    try:
+        template["parameters"]["poolName"]["defaultValue"] = pool_id
+    except KeyError:
+        pass
```

Comment:
I don't feel like thses are Redundant 

def set_template_name(template, pool_id):
    try:
        template["parameters"]["poolName"]["defaultValue"] = pool_id
    except KeyError:
        pass
    try:
        template["parameters"]["poolId"]["defaultValue"] = pool_id
    except KeyError:
        pass
    try:
        template["parameters"]["poolId"]["value"] = pool_id
    except KeyError:
        pass

vs this are very different 
def set_template_name(template, pool_id):
    try:
        template["parameters"]["poolId"]["value"] = pool_id
        template["parameters"]["poolName"]["defaultValue"] = pool_id
        template["parameters"]["poolId"]["defaultValue"] = pool_id
    except KeyError:
        pass

The point of these methods is to set a single value in the template, if that value isn't in there in the template then that means we failover and try and set a different value. From my understanding, the second example would fail if the first set fails and it won't proceed to the second set.